torchao/prototype/mx_formats/mx_tensor.py

```diff
@@ -559,10 +559,17 @@ def __repr__(self):
     @classmethod
     def __torch_dispatch__(cls, func, types, args, kwargs=None):
         # avoid circular dependency
+        from torchao.prototype.mx_formats.mx_funcs import MX_FUNC_TABLE
         from torchao.prototype.mx_formats.mx_ops import MX_OPS_TABLE
 
         if func in MX_OPS_TABLE:
-            return MX_OPS_TABLE[func](func, args, kwargs)
+            return MX_OPS_TABLE[func](func, types, args, kwargs)
+
+        # TODO AO BASE_TENSOR doesn't respect dispatch and function modes
```

Comment:
is this comment up to date? `MXTensor` does not inherit from AOBaseTensor at the moment

Reply:
yeah this is still relevant because we are using the Linear helper subclass which does inherit from AO base

Reply:
Left a better comment